src/ajax/xhr.js

```diff
@@ -19,7 +19,19 @@ jQuery.ajaxSettings.xhr = window.ActiveXObject !== undefined ?
 			// and http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9
 			// Although this check for six methods instead of eight
 			// since IE also does not support "trace" and "connect"
-			/^(get|post|head|put|delete|options)$/i.test( this.type ) &&
+			//
+			// Support: IE 10-11
+			// IE seems to error on cross-domain PATCH requests when ActiveX XHR
+			// is used. Use the following strategy:
+			// 1. If request is local, use the ActiveX XHR.
+			// 2. In IE9+ always use the native XHR.
+			// 3. If IE8 and the request type is not included in the following list,
+			//    use the ActiveX XHR.
+			// 4. Otherwise use the native XHR.
+			// jshint -W018
+			( !( document.documentMode < 9 ) ||
```

Comment:
It's a browser sniff, not UA sniff (changes in UA won't make this code behave differently). Also, this will not catch any new browsers but only old, specific ones (i.e. only IE8 or newer IE in IE8 document mode) so this will not have any unwanted effect on new browsers.

But if you have any idea how to test for this specific issue, I'm listening. :) I just don't think it's possible to test for such weird cases in JS code, especially without seriously impacting performance.